joblib/numpy_pickle.py

```diff
@@ -333,44 +338,32 @@ def load_build(self):
         dispatch[pickle.BUILD] = load_build
 
 
-class ZipNumpyUnpickler(NumpyUnpickler):
-    """A subclass of our Unpickler to unpickle on the fly from
-    compressed storage."""
-
-    def __init__(self, filename, file_handle):
-        NumpyUnpickler.__init__(self, filename,
-                                file_handle,
-                                mmap_mode=None)
-
-    def _open_pickle(self, file_handle):
-        return BytesIO(read_zfile(file_handle))
-
-
 ###############################################################################
 # Utility functions
 
-def dump(value, filename, compress=0, cache_size=100, protocol=None):
-    """Fast persistence of an arbitrary Python object into one or multiple
-    files, with dedicated storage for numpy arrays.
+def dump(value, filename, compress=0, protocol=None, cache_size=None):
+    """Persist an arbitrary Python object into one file.
 
     Parameters
     -----------
     value: any Python object
         The object to store to disk
-    filename: string or pathlib.Path
+    filename: str or pathlib.Path
         The path of the file in which it is to be stored
```

Comment:
I would also make it such as if `filename` is a string and `filename.endswith('.gz') and compress is None`, then we would automatically set `compress=('gzip', 3)`, and similarly for filenames ending in ".bz2" and ".xz".

What do you think @aabadie @GaelVaroquaux and @lesteve?